src/test/java/com/alvarium/exampleapp/observers/TransitionObserverTest.java

```diff
@@ -0,0 +1,67 @@
+package com.alvarium.exampleapp.observers;
+
+import com.alvarium.Sdk;
+import com.alvarium.exampleapp.SampleData;
+import com.alvarium.sign.KeyInfo;
+import com.alvarium.sign.SignType;
+import com.alvarium.utils.PropertyBag;
+
+import org.junit.Test;
+
+import io.reactivex.rxjava3.core.Observer;
+
+public class TransitionObserverTest {
+
+  @Test
+  public void transitionObserverShouldTransitAndPublish()throws Exception{
+    final String privateKeyPath = "./src/test/java/com/alvarium/exampleapp/private.key";
+    final KeyInfo keyInfo = new KeyInfo(privateKeyPath, SignType.Ed25519);
+    final SampleData sampleData = new SampleData(keyInfo);
+
+    final Channel<SampleData> source = new Channel<SampleData>();
+    
+    final Sdk sdk = new Sdk() {
```

Comment:
What you have here is essentially a mock object that's [duplicated](https://github.com/project-alvarium/example-java/blob/d995b6ce4970f5361a68987ef24e53325a608b28/src/test/java/com/alvarium/exampleapp/observers/CreationObserverTest.java#L24) from the CreationObserverTest. Have you looked at unit test frameworks such as JUnit or Mockito for the purpose of mocking out these interface methods?

Also the text printed to the console should have a prefix indicating which method is writing the output for clarity.